cpp/src/round/round.cu

```diff
@@ -252,13 +288,29 @@ std::unique_ptr<column> round_with(column_view const& input,
     result_type, input.size(), copy_bitmask(input, stream, mr), input.null_count(), stream, mr);
 
   auto out_view = result->mutable_view();
-  Type const n  = std::pow(10, std::abs(decimal_places + input.type().scale()));
 
-  thrust::transform(rmm::exec_policy(stream),
-                    input.begin<Type>(),
-                    input.end<Type>(),
-                    out_view.begin<Type>(),
-                    FixedPointRoundFunctor{n});
+  constexpr int max_precision = [] {
+    if constexpr (std::is_same_v<T, numeric::decimal32>) return 9;
+    if constexpr (std::is_same_v<T, numeric::decimal64>) return 18;
+    return 38;
+  }();
```

Comment:
This may work here
```suggestion
  constexpr int max_precision = cuda::std::numeric_limits<T::rep>::digits10;
```
You may need to add `#include <cuda/std/limits>`

Reply:
Yep, please use the `digits10`.

Reply:
Done.